util/misc.py

```diff
@@ -175,19 +180,38 @@ def run(args, stdin=None, stdout=None, stderr=None, shell=False,
 
 def run_and_print(args, stdout=None, stderr=None,
                   stdin=None, shell=False, env=None, cwd=None,
-                  timeout=None, silent=False, buffered=False, check=False):
+                  timeout=None, silent=False, buffered=False, check=False,
+                  loglevel=None):
     '''Capture stdout+stderr and print.
 
     This is useful for nose, which has difficulty capturing stdout of
     subprocess invocations.
     '''
-
+    if loglevel:
+        silent = False
     if not buffered:
-        result = run(args, stdin=stdin, stdout=subprocess.PIPE,
-                     stderr=subprocess.STDOUT, env=env, cwd=cwd,
-                     timeout=timeout, check=check)
-        if not silent:
-            print(result.stdout.decode('utf-8'))
+        if check and not silent:
+            try:
+                result = run(args, stdin=stdin, stdout=subprocess.PIPE,
+                           stderr=subprocess.STDOUT, env=env, cwd=cwd,
+                           timeout=timeout, check=check)
+            except subprocess.CalledProcessError as e:
+                if loglevel:
+                    logging.log(loglevel, result.stdout.decode('utf-8'))
```

Comment:
Interesting... instead of using the root level logger (logging.log), can you do what the other toplevel scripts do and put a
```
log = logging.getLogger(__name__)
```
global variable at the top of the file and use that "log" variable here?

Also separately, you'll want to merge origin/master back in here, as Chris has independently fixed the run-twice issue there.